apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/UserInfoController.java

```diff
@@ -32,34 +38,35 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-import java.util.List;
-
 @RestController
 public class UserInfoController {
 
   private final UserInfoHolder userInfoHolder;
   private final LogoutHandler logoutHandler;
   private final UserService userService;
+  private final AuthUserPasswordChecker passwordChecker;
 
   public UserInfoController(
       final UserInfoHolder userInfoHolder,
       final LogoutHandler logoutHandler,
-      final UserService userService) {
+      final UserService userService,
+      final AuthUserPasswordChecker passwordChecker) {
     this.userInfoHolder = userInfoHolder;
     this.logoutHandler = logoutHandler;
     this.userService = userService;
+    this.passwordChecker = passwordChecker;
   }
 
-
   @PreAuthorize(value = "@permissionValidator.isSuperAdmin()")
   @PostMapping("/users")
   public void createOrUpdateUser(@RequestBody UserPO user) {
     if (StringUtils.isContainEmpty(user.getUsername(), user.getPassword())) {
       throw new BadRequestException("Username and password can not be empty.");
     }
+    CheckResult pwdCheckRes;
+    if (!(pwdCheckRes = passwordChecker.checkWeakPassword(user.getPassword())).isSuccess()) {
+      throw new BadRequestException(pwdCheckRes.getMessage());
+    }
```

Comment:
Is there a reason why you assign `pwdCheckRes` inside the `if` and not before when you declare the var `pwdCheckRes`?

Otherwise to make the code more readable I would suggest to change it to 
```java
CheckResult pwdCheckRes = passwordChecker.checkWeakPassword(user.getPassword());
if (!pwdCheckRes.isSuccess()) {
  throw new BadRequestException(pwdCheckRes.getMessage());
}
```

Reply:
Ok, done.